Makefile

```diff
@@ -383,7 +383,10 @@ validateXML: createXMLExamples
 	xmllint --noout --relaxng cppcheck-errors.rng /tmp/example.xml
 
 checkCWEEntries: /tmp/errorlist.xml
-	./tools/listErrorsWithoutCWE.py -F /tmp/errorlist.xml
+	$(eval PYTHON_INTERPRETER := $(if $(PYTHON_INTERPRETER),$(PYTHON_INTERPRETER),$(shell which python3)))
```

Comment:
I don't have a very strong opinion but I think we can require python3 for this script. It's not something a normal user will run.

Reply:
As outlined in #3596 I think we can completely drop Python 2.7. 

Also after this is merged we can hopefully merge #3596 as well and we default to Python 3.x everywhere.

I would give it a release cycle to sink in, adding deprecation warnings to the build at the start of the next dev cycle (2.9) and drop it in the one afterward (2.10).

Reply:
but I wonder what the motivation is to set PYTHON_INTERPRETER here. isn't it enough to use the shebang as we used to?

Reply:
You can specify the interpreter when running `make`:

`PYTHON_INTERPRETER=/usr/local/bin/python3.11 make`

So this is done for consistency. It will mainly be necessary for testing and MinGW on Windows if the interpreter is not in the `PATH`.

Before this change it was done once globally but only with `MATCHCOMPILER=yes`. Since that might not be set and we don't want to do it unconditionally I had to replicate this since I was not able to do it in conditional code.

Reply:
As far as I know, until now those that used `checkCweEntries` was satisfied with the shebang path. It's a script that is used very rarelly.

Reply:
It also fails on platforms like Ubuntu 22.04 where `python` not longer exists by default and the shebang is not changed yet. The other way around it would fail if no Python 3 is installed by default like CentOS after the shebang is changed. So this is a required change to make the CI work.